### PR TITLE
Replace Supplier by LoadingCache for caching PathFilter when loading partition splits of HUDI tables

### LINKED ISSUE #16598: UnknownHostException With Presto Query Hudi Table
When HA is enabled on the NameNode, an UnknownHostException: nameservice1 occurs when Presto query the Hudi table。But querying Hive table is normal。

PrestoDB Version：0.258
Hudi Version：0.9

hive.properties

> connector.name=hive-hadoop2
hive.metastore.uri=thrift://xxxx.glprop.com:9083
hive.config.resources=/etc/hadoop/conf/hdfs-site.xml,/etc/hadoop/conf/core-site.xml
hive.metastore.authentication.type=KERBEROS
hive.metastore.service.principal=hive/_HOST@GLPROP.COM
hive.metastore.client.principal=bigdata@GLPROP.COM
hive.metastore.client.keytab=/etc/security/keytab/bigdata.keytab
hive.hdfs.authentication.type=KERBEROS
hive.hdfs.presto.principal=bigdata@GLPROP.COM
hive.hdfs.presto.keytab=/etc/security/keytab/bigdata.keytab

Exception:
2021-08-12T14:37:43.410+0800    ERROR   hive-hive-1     org.apache.hudi.hadoop.HoodieROTablePathFilter  Error checking path :hdfs://nameservice1/hudi/ods_inout_record_cow/20210809/.hoodie_partition_metadata, under folder: null
java.lang.IllegalArgumentException: java.net.UnknownHostException: nameservice1
        at org.apache.hadoop.security.SecurityUtil.buildTokenService(SecurityUtil.java:378)
        at org.apache.hadoop.hdfs.NameNodeProxies.createNonHAProxy(NameNodeProxies.java:320)
        at org.apache.hadoop.hdfs.NameNodeProxies.createProxy(NameNodeProxies.java:176)
        at org.apache.hadoop.hdfs.DFSClient.<init>(DFSClient.java:678)
        at org.apache.hadoop.hdfs.DFSClient.<init>(DFSClient.java:619)
        at org.apache.hadoop.hdfs.DistributedFileSystem.initialize(DistributedFileSystem.java:149)
        at org.apache.hadoop.fs.PrestoFileSystemCache.createFileSystem(PrestoFileSystemCache.java:149)
        at org.apache.hadoop.fs.PrestoFileSystemCache.getInternal(PrestoFileSystemCache.java:89)
        at org.apache.hadoop.fs.PrestoFileSystemCache.get(PrestoFileSystemCache.java:62)
        at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:373)
        at org.apache.hadoop.fs.Path.getFileSystem(Path.java:295)
        at org.apache.hudi.hadoop.HoodieROTablePathFilter.accept(HoodieROTablePathFilter.java:106)
        at com.facebook.presto.hive.util.HiveFileIterator.lambda$getLocatedFileStatusRemoteIterator$0(HiveFileIterator.java:103)
        at com.google.common.collect.Iterators$5.computeNext(Iterators.java:639)
        at com.google.common.collect.AbstractIterator.tryToComputeNext(AbstractIterator.java:141)
        at com.google.common.collect.AbstractIterator.hasNext(AbstractIterator.java:136)
        at com.facebook.presto.hive.util.HiveFileIterator.computeNext(HiveFileIterator.java:69)
        at com.facebook.presto.hive.util.HiveFileIterator.computeNext(HiveFileIterator.java:40)
        at com.google.common.collect.AbstractIterator.tryToComputeNext(AbstractIterator.java:141)
        at com.google.common.collect.AbstractIterator.hasNext(AbstractIterator.java:136)
        at java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1811)
        at java.util.stream.StreamSpliterators$WrappingSpliterator.lambda$initPartialTraversalState$0(StreamSpliterators.java:294)
        at java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.fillBuffer(StreamSpliterators.java:206)
        at java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.doAdvance(StreamSpliterators.java:161)
        at java.util.stream.StreamSpliterators$WrappingSpliterator.tryAdvance(StreamSpliterators.java:300)
        at java.util.Spliterators$1Adapter.hasNext(Spliterators.java:681)
        at com.facebook.presto.hive.BackgroundHiveSplitLoader.loadSplits(BackgroundHiveSplitLoader.java:195)
        at com.facebook.presto.hive.BackgroundHiveSplitLoader.access$300(BackgroundHiveSplitLoader.java:40)
        at com.facebook.presto.hive.BackgroundHiveSplitLoader$HiveSplitLoaderTask.process(BackgroundHiveSplitLoader.java:121)
        at com.facebook.presto.hive.util.ResumableTasks.safeProcessTask(ResumableTasks.java:47)
        at com.facebook.presto.hive.util.ResumableTasks.access$000(ResumableTasks.java:20)
        at com.facebook.presto.hive.util.ResumableTasks$1.run(ResumableTasks.java:35)
        at com.facebook.airlift.concurrent.BoundedExecutor.drainQueue(BoundedExecutor.java:78)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.net.UnknownHostException: nameservice1
        ... 36 more

### PR DESCRIPTION
**Pull request by imjalpreet**
Currently, PathFilter is configured with default configuration which can lead to issues when PrestoFileSystemCache is refreshed. Now, PathFilter will be initialized with custom updated configuration object.

Fixes #16598 

```
== RELEASE NOTES ==

Hive Changes
* ...
* ...
```

### CODE REVIEW

**Comment by dborkar** [comment]
@tdcmeehan  would be great to get this merged. several folks waiting for it.

File: presto-hive/src/main/java/com/facebook/presto/hive/StoragePartitionLoader.java [hunk 1]
```diff
@@ -137,7 +139,8 @@ public StoragePartitionLoader(
         this.hdfsContext = new HdfsContext(session, table.getDatabaseName(), table.getTableName(), table.getStorage().getLocation(), false);
         this.fileIterators = requireNonNull(fileIterators, "fileIterators is null");
         this.schedulerUsesHostAddresses = schedulerUsesHostAddresses;
-        this.hoodiePathFilterSupplier = Suppliers.memoize(HoodieROTablePathFilter::new)::get;
+        this.hoodiePathFilterLoadingCache = CacheBuilder.newBuilder().build(CacheLoader.from(
```
Comment (tdcmeehan) [inline presto-hive/src/main/java/com/facebook/presto/hive/StoragePartitionLoader.java:142]: Is it possible for this cache to grow unbounded?

**Review by tdcmeehan — COMMENTED** [review]
LGTM, but please make sure the commit message follows our guidelines: https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md#pull-request-checklist

In particular, the title is too long, and the body of the commit message is also too long.

I also have a question on the nature of the cache and whether this could grow unbounded.

File: presto-hive/src/main/java/com/facebook/presto/hive/StoragePartitionLoader.java (hunk 1 above)
Comment (imjalpreet) [reply presto-hive/src/main/java/com/facebook/presto/hive/StoragePartitionLoader.java:142]: Ideally it won't but as a preventive measure I will make a change to limit the maximum size of the cache.

**Review by tdcmeehan — APPROVED** [review]

**Pull request merged**